It looks like maybe the pid file is an actual problem in the flake here

```
>           assert os.path.exists(os.path.join(temp_dir, 'pid'))
E           AssertionError: assert False
E            +  where False = <function exists at 0x7fe35f4f45e0>('/tmp/tmp4cg14jf2/pid')
E            +    where <function exists at 0x7fe35f4f45e0> = <module 'posixpath' from '/home/zuul/src/github.com/ansible/ansible-runner/.tox/ansible28/lib64/python3.8/posixpath.py'>.exists
E            +      where <module 'posixpath' from '/home/zuul/src/github.com/ansible/ansible-runner/.tox/ansible28/lib64/python3.8/posixpath.py'> = os.path
E            +    and   '/tmp/tmp4cg14jf2/pid' = <function join at 0x7fe35f4f4e50>('/tmp/tmp4cg14jf2', 'pid')
E            +      where <function join at 0x7fe35f4f4e50> = <module 'posixpath' from '/home/zuul/src/github.com/ansible/ansible-runner/.tox/ansible28/lib64/python3.8/posixpath.py'>.join
E            +        where <module 'posixpath' from '/home/zuul/src/github.com/ansible/ansible-runner/.tox/ansible28/lib64/python3.8/posixpath.py'> = os.path
```

--- a/test/integration/conftest.py
+++ b/test/integration/conftest.py
@@ -1,10 +1,13 @@
+import os
+import shutil
+
 import pytest
 import pexpect
 from ansible_runner.runner_config import RunnerConfig
 
 
 @pytest.fixture(scope='function')
-def rc(request, tmpdir):
+def rc(tmpdir):
     rc = RunnerConfig(str(tmpdir))
     rc.suppress_ansible_output = True
     rc.expect_passwords = {
@@ -13,9 +16,9 @@ def rc(request, tmpdir):
     }
     rc.cwd = str(tmpdir)
     rc.env = {}
-    rc.job_timeout = 2
+    rc.job_timeout = 10
     rc.idle_timeout = 0
-    rc.pexpect_timeout = .1
+    rc.pexpect_timeout = 2.
     rc.pexpect_use_poll = True
     return rc
 
@@ -39,7 +42,7 @@ def container_runtime_available():
 
 # TODO: determine if we want to add docker / podman
 # to zuul instances in order to run these tests
-@pytest.fixture(scope="session", autouse=True)
+@pytest.fixture(scope="session")
 def container_runtime_installed():
     import subprocess
 
@@ -50,3 +53,21 @@ def container_runtime_installed():
         except FileNotFoundError:
             pass
     pytest.skip('No container runtime is available.')
+
+
+@pytest.fixture(scope='session')
+def clear_integration_artifacts(request):
+    '''Fixture is session scoped to allow parallel runs without error
+    '''
+    if 'PYTEST_XDIST_WORKER' in os.environ:
+        # we never want to clean artifacts if running parallel tests
+        # because we cannot know when all processes are finished and it is
+        # safe to clean up
+        return
+
+    def rm_integration_artifacts():
+        path = "test/integration/artifacts"
+        if os.path.exists(path):
+            shutil.rmtree(path)
+
+    request.addfinalizer(rm_integration_artifacts)

--- a/test/integration/test_events.py
+++ b/test/integration/test_events.py
@@ -3,12 +3,10 @@ import tempfile
 from distutils.spawn import find_executable
 import json
 import os
-import shutil
 
 from ansible_runner import run, run_async
 
 
-@pytest.mark.serial
 @pytest.mark.parametrize('containerized', [True, False])
 def test_basic_events(containerized, container_runtime_available, is_pre_ansible28, is_run_async=False, g_facts=False):
     if containerized and not container_runtime_available:
@@ -60,7 +58,6 @@ def test_basic_events(containerized, container_runtime_available, is_pre_ansible
     assert "event_data" in okay_event and len(okay_event['event_data']) > 0
 
 
-@pytest.mark.serial
 @pytest.mark.parametrize('containerized', [True, False])
 def test_async_events(containerized, container_runtime_available, is_pre_ansible28):
     test_basic_events(containerized, container_runtime_available, is_pre_ansible28, is_run_async=True,g_facts=True)
@@ -134,22 +131,18 @@ def test_playbook_on_stats_summary_fields(rc, is_pre_ansible28):
     assert set(fields) >= set(EXPECTED_SUMMARY_FIELDS)
 
 
-@pytest.mark.serial
-def test_include_role_events():
-    try:
-        r = run(
-            private_data_dir=os.path.abspath('test/integration'),
-            playbook='use_role.yml'
-        )
-        role_events = [event for event in r.events if event.get('event_data', {}).get('role', '') == "benthomasson.hello_role"]
-        assert 'runner_on_ok' in [event['event'] for event in role_events]
-        for event in role_events:
-            event_data = event['event_data']
-            assert not event_data.get('warning', False)  # role use should not contain warnings
-            if event['event'] == 'runner_on_ok':
-                assert event_data['res']['msg'] == 'Hello world!'
-    finally:
-        shutil.rmtree('test/integration/artifacts')
+def test_include_role_events(clear_integration_artifacts):
+    r = run(
+        private_data_dir=os.path.abspath('test/integration'),
+        playbook='use_role.yml'
+    )
+    role_events = [event for event in r.events if event.get('event_data', {}).get('role', '') == "benthomasson.hello_role"]
+    assert 'runner_on_ok' in [event['event'] for event in role_events]
+    for event in role_events:
+        event_data = event['event_data']
+        assert not event_data.get('warning', False)  # role use should not contain warnings
+        if event['event'] == 'runner_on_ok':
+            assert event_data['res']['msg'] == 'Hello world!'
 
 
 @pytest.mark.skipif(find_executable('cgexec') is None,

--- a/test/integration/test_interface.py
+++ b/test/integration/test_interface.py
@@ -43,7 +43,6 @@ def get_env_data(res):
         raise RuntimeError('Count not find look_at_environment task from playbook')
 
 
-@pytest.mark.serial
 def test_env_accuracy(request, printenv_example):
     os.environ['SET_BEFORE_TEST'] = 'MADE_UP_VALUE'
 

--- a/test/integration/test_main.py
+++ b/test/integration/test_main.py
@@ -104,7 +104,6 @@ def test_module_run_debug():
             shutil.rmtree('./ping')
 
 
-@pytest.mark.serial
 def test_module_run_clean():
     with temp_directory() as temp_dir:
         rc = main(['run', '-m', 'ping',
@@ -113,13 +112,12 @@ def test_module_run_clean():
     assert rc == 0
 
 
-def test_role_run(skipif_pre_ansible28):
+def test_role_run(skipif_pre_ansible28, clear_integration_artifacts):
     rc = main(['run', '-r', 'benthomasson.hello_role',
                '--hosts', 'localhost',
                '--roles-path', 'test/integration/roles',
                "test/integration"])
     assert rc == 0
-    ensure_removed("test/integration/artifacts")
 
 
 def test_role_run_abs():
@@ -131,17 +129,14 @@ def test_role_run_abs():
     assert rc == 0
 
 
-def test_role_logfile(skipif_pre_ansible28):
-    try:
-        rc = main(['run', '-r', 'benthomasson.hello_role',
-                   '--hosts', 'localhost',
-                   '--roles-path', 'test/integration/project/roles',
-                   '--logfile', 'new_logfile',
-                   'test/integration'])
-        assert os.path.exists('new_logfile')
-        assert rc == 0
-    finally:
-        ensure_removed("test/integration/artifacts")
+def test_role_logfile(skipif_pre_ansible28, clear_integration_artifacts):
+    rc = main(['run', '-r', 'benthomasson.hello_role',
+               '--hosts', 'localhost',
+               '--roles-path', 'test/integration/project/roles',
+               '--logfile', 'new_logfile',
+               'test/integration'])
+    assert os.path.exists('new_logfile')
+    assert rc == 0
 
 
 def test_role_logfile_abs():
@@ -175,15 +170,13 @@ def test_role_bad_project_dir():
         ensure_removed("new_logfile")
 
 
-@pytest.mark.serial
-def test_role_run_clean(skipif_pre_ansible28):
+def test_role_run_clean(skipif_pre_ansible28, clear_integration_artifacts):
 
     rc = main(['run', '-r', 'benthomasson.hello_role',
                '--hosts', 'localhost',
                '--roles-path', 'test/integration/roles',
                "test/integration"])
     assert rc == 0
-    ensure_removed("test/integration/artifacts")
 
 
 def test_role_run_cmd_line_abs():
@@ -195,14 +188,13 @@ def test_role_run_cmd_line_abs():
     assert rc == 0
 
 
-def test_role_run_artifacts_dir(skipif_pre_ansible28):
+def test_role_run_artifacts_dir(skipif_pre_ansible28, clear_integration_artifacts):
     rc = main(['run', '-r', 'benthomasson.hello_role',
                '--hosts', 'localhost',
                '--roles-path', 'test/integration/roles',
                '--artifact-dir', 'otherartifacts',
                "test/integration"])
     assert rc == 0
-    ensure_removed("test/integration/artifacts")
 
 
 def test_role_run_artifacts_dir_abs(skipif_pre_ansible28):
@@ -295,7 +287,6 @@ def test_role_start():
         p.join()
 
 
-@pytest.mark.serial
 def test_playbook_start(skipif_pre_ansible28):
 
     inv = 'inventory/localhost'

